File: scheduler/core/src/main/java/io/orkes/conductor/scheduler/service/SchedulerBulkService.java
package io.orkes.conductor.scheduler.service;

import java.util.List;

import org.springframework.validation.annotation.Validated;

import com.netflix.conductor.common.model.BulkResponse;

import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.Size;

@Validated
public interface SchedulerBulkService {

    int MAX_REQUEST_ITEMS = 1000;

    BulkResponse pauseSchedules(
            @NotEmpty(message = "Schedule names list cannot be null.")
                    @Size(
                            max = MAX_REQUEST_ITEMS,
                            message =
                                    "Cannot process more than {max} schedules. Please use multiple requests.")
                    List<String> scheduleNames);

    BulkResponse resumeSchedules(
            @NotEmpty(message = "Schedule names list cannot be null.")
                    @Size(
                            max = MAX_REQUEST_ITEMS,
                            message =
                                    "Cannot process more than {max} schedules. Please use multiple requests.")
                    List<String> scheduleNames);
}

File: scheduler/core/src/main/typescript/io/orkes/conductor/scheduler/service/SchedulerBulkService.ts
import { BulkResponse } from 'com/netflix/conductor/common/model/BulkResponse';

export interface SchedulerBulkService {
    pauseSchedules(scheduleNames: Array<string>): BulkResponse;

    resumeSchedules(scheduleNames: Array<string>): BulkResponse;
}

export const MAX_REQUEST_ITEMS = 1000;
